src/main/scala/cromwell/binding/Call.scala

```diff
@@ -1,11 +1,56 @@
 package cromwell.binding
 
 import cromwell.binding.types.WdlType
-import cromwell.binding.values.WdlValue
 import cromwell.parser.AstTools
+import cromwell.parser.WdlParser.{SyntaxError, Terminal, Ast}
+import cromwell.parser.AstTools.EnhancedAstNode
 
 import scala.util.{Success, Try}
 
+object Call {
```

Comment:
Yeah, it's just moving the stuff around to put the logic of construction/validation of the component types to the type